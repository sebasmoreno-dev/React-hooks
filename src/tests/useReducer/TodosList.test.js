import React from 'react';
import { shallow } from 'enzyme';
import { TodoList } from './../../components/08-useReducer/TodoList';
import { demoTodos } from './../fixtures/demoTodos';
import './../../setupTest';



describe('Pruebas en TodoList', () => {

  const handleDelete = jest.fn();
  const handleToggle = jest.fn();

  const wrapper = shallow(
    <TodoList
      todos={ demoTodos }
      handleDelete={handleDelete}
      handleToggle={handleToggle}
      />
    );


  test('Debe de mostrar el componente correctamente', () => {

    expect(wrapper).toMatchSnapshot();
  });


  test('Debe tener los TodoListItem', () => {

    expect(wrapper.find('TodoListItem').length).toBe(demoTodos.length);

    expect(wrapper.find('TodoListItem').at(0).prop('handleDelete')).toEqual( expect.any(Function) );
  });


});