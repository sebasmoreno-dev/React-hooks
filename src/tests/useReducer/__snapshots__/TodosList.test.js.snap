// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en TodoList Debe de mostrar el componente correctamente 1`] = `
<ul
  className="list-group list-group-flush"
>
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    index={0}
    key="1"
    todo={
      Object {
        "desc": "Hacer ejercicio",
        "done": false,
        "id": 1,
      }
    }
  />
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    index={1}
    key="2"
    todo={
      Object {
        "desc": "Hacer ejercicio",
        "done": false,
        "id": 2,
      }
    }
  />
</ul>
`;

exports[`Pruebas en todosList Debe de mostrar el componente correctamente 1`] = `
<ul
  className="list-group list-group-flush"
>
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    index={0}
    key="1"
    todo={
      Object {
        "desc": "Hacer ejercicio",
        "done": false,
        "id": 1,
      }
    }
  />
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToggle={[MockFunction]}
    index={1}
    key="2"
    todo={
      Object {
        "desc": "Hacer ejercicio",
        "done": false,
        "id": 2,
      }
    }
  />
</ul>
`;
